### ISSUE TITLE
 How to understand “each mini-batch Yi always contains these five training samples of this class” and how to change it.

### ISSUE THREAD

**Issue by Torch-HXM** [swiftest/RSSGL#1]
Your paper is very well written and I appreciate your approach, but I still have some questions.
As it is said in your paper : “each mini-batch Yi always contains these five training samples of this class”.
![image](https://github.com/swiftest/RSSGL/assets/88866541/353157bf-5be0-4dec-8db5-d63455955fa8)
Does it means that each batch from the dataloader contains these five training samples? How can I disable this feature in your code? I just want a normal random choose.
Waiting for your replay, thank you.

**Comment by swiftest** [comment]
@Torch-HXM 
Thank you for your interest in our research.  If you want to use a normal random choose, first change the batch_size setting in the configuration file (i.e. './configs/RSSGL/RSSGL_.....py'), next you need to modify the minibatch_sample function in the base module in the data folder. 

The batch_size in the global learning framework should not be set too small, according to your requirements, the modified code is shown below.

def minibatch_sample(gt_mask: np.ndarray, train_indicator: np.ndarray, batch_size):
     """
    Args:
        gt_mask: 2-D array of shape [height, width]
        train_indicator: 2-D array of shape [height, width]
        minibatch_size:
    Returns:
    """

    inds = np.where(train_indicator.ravel() == 1)[0]
    np.random.shuffle(inds)
    train_inds_list = []
    cnt = 0

    while True:
        train_inds = np.zeros_like(train_indicator).ravel()
        left = cnt * batch_size  # Suppose batch_size is set to 100, there 529 training samples for IP data set.
        if left >= len(inds):
            break
        right = min((cnt + 1) * batch_size, len(inds))
        fetch_inds = inds[left:right]
        train_inds[fetch_inds] = 1
        train_inds_list.append(train_inds.reshape(train_indicator.shape))
        cnt += 1

    return train_inds_list


If you have any further questions, you can email me at wanghenghrbeu@hrbeu.edu.cn.
